File: 1-Basics/1.1-Patterns/Patterns_python/7-StarPyramid.py
n = int(input("Enter Row Size N : "))

for i in range(1,n+1):
    # left space
    for j in range(n-i):
        print(" ", end=" ")
    
    # star 
    for l in range(1,(i*2)):
        print("*",end=" ")

    print('\n')

File: 1-Basics/1.1-Patterns/Patterns_cpp/7-StarPyramid.cpp
#include<bits/stdc++.h>
using namespace std;

// - - 1 - -
// - 1 2 3 -
// 1 2 3 4 5 

void pattern(int n){
    for(int i=1; i<=n; i++){

        // left space 
        for(int j=n-i; j>0; j--){
            cout<<" ";
        }

        // star 
        for(int k=1; k<=(i*2)-1; k++){
            cout<<"*";
        }

        cout<<endl;
    }
}

int main(){
    cout<<"Enter Row Size N : ";
    int n;
    cin>>n;

    pattern(n);

    return 0;
}